granulate_utils/linux/process.py

```diff
@@ -63,12 +64,21 @@ def get_mapped_dso_elf_id(process: psutil.Process, dso_part: str) -> Optional[st
         return None
 
 
+def read_proc_file(process: psutil.Process, name: str) -> bytes:
```

Comment:
gProfiler's verifies that `process` is still running afterwards (protects from PID reuse). Let's add this verification here.

Also - please open the relevant PR in gProfiler so it uses your new API, to avoid keeping 2 copies of the function.